src/extension.ts

```diff
@@ -117,20 +120,24 @@ function initIntelliSense(context: vscode.ExtensionContext): void {
   );
   const diagnosticProvider = new DigitalTwinDiagnosticProvider();
   const activeTextEditor: vscode.TextEditor | undefined = vscode.window.activeTextEditor;
-  if (activeTextEditor) {
-    diagnosticProvider.updateDiagnostics(activeTextEditor.document, diagnosticCollection);
+  if (activeTextEditor && isJsonFile(activeTextEditor.document)) {
```

Comment:
>isJsonFile(activeTextEditor.document)) [](start = 26, length = 38)

Why enable diagnostic if a JSON file is detected rather than a DTDL file?

Reply:
Since DTDL file is basically a JSON file, narrow down the file type to reduce unnecessary calls on other file types. For Auto-compeletion and Hover, we also use document selector for this purpose.